polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1242,7 +1253,8 @@ async fn handle_network_msg<Context>(
 		},
 		OurViewChange(view) => {
 			gum::trace!(target: LOG_TARGET, ?view, "Own view change");
-			handle_our_view_change(ctx, state, view).await?;
+			handle_our_view_change(ctx, runtime, state, view).await?;
+			connect_to_validators(ctx, &state.implicit_view, &state.per_relay_parent).await;
```

Comment:
if we do this on our view change, then the delayed call of this function triggered by RECONNECT_AFTER_LEAF_TIMEOUT does not make much sense any more.
I think we could remove that.

Let's also remove all comments mentioning `ValidatorGroupsBuffer`

Reply:
> then the delayed call of this function triggered by RECONNECT_AFTER_LEAF_TIMEOUT 

That't more or less to force the reconnect if this call failed to connect the validators, why wouldn't still be needed ? You mean because we do it anyways every 6 seconds ?